src/Ilios/CoreBundle/Entity/User.php

```diff
@@ -1325,23 +1325,52 @@ public function getEncoderName()
     }
     
     /**
-     * Get all the schools an user is affiliated with
-     * This is used in granting view permissions for faculty
-     * in schools that are a users secondary cohort
+     * Get all the schools an user is affiliated with so we can match
+     * permissions.
      *
      * @return ArrayCollection[School]
      */
     public function getAllSchools()
     {
-        $undeletedCohorts = $this->getCohorts()->filter(function (CohortInterface $cohort) {
+        $cohortSchools = $this->getCohorts()->filter(function (CohortInterface $cohort) {
             return !$cohort->isDeleted();
-        });
-        $schools = $undeletedCohorts->map(function (CohortInterface $cohort) {
+        })->map(function (CohortInterface $cohort) {
             return $cohort->getProgramYear()->getProgram()->getSchool();
         });
-        $schools->add($this->getSchool());
-        
-        
+
+        $directedCourseSchools = $this->getDirectedCourses()->map(function (CourseInterface $course) {
+            return $course->getSchool();
+        });
+
+        $learnerGroupSchools = $this->getLearnerGroups()->map(function (LearnerGroupInterface $lg) {
+            return $lg->getCohort()->getProgramYear()->getProgram()->getSchool();
+        });
+
+        $instructedLgSchools = $this->getInstructedLearnerGroups()->map(function (LearnerGroupInterface $lg) {
+            return $lg->getCohort()->getProgramYear()->getProgram()->getSchool();
+        });
+
+        $instGroupSchools = $this->getInstructorGroups()->map(function (InstructorGroupInterface $ig) {
+            return $ig->getSchool();
+        });
+
+        $insIlmSchools = $this->getInstructorIlmSessions()->map(function (IlmSessionInterface $ilm) {
+            return $ilm->getSession()->getCourse()->getSchool();
+        });
+
+        $allSchools = array_merge(
+            $cohortSchools->toArray(),
+            $directedCourseSchools->toArray(),
+            $learnerGroupSchools->toArray(),
+            $instructedLgSchools->toArray(),
+            $instGroupSchools->toArray(),
+            $insIlmSchools->toArray()
+        );
+        $allSchools[] = $this->getSchool();
+
+
```

Comment:
Please add a call to `array_filter` here to get rid of any nulls that may have crept into `$allSchools` at this point.